### ISSUE TITLE
Autoupgrade 8.1.0 to 8.1.1 error database update

### ISSUE THREAD

**Issue by eveneme** [PrestaShop/PrestaShop#33564]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

`[INTERNAL] src/PrestaShopBundle/Command/UpdateSchemaCommand.php line 141 - TypeError: Unsupported operand types: int + Doctrine\DBAL\ForwardCompatibility\Result #0 src/PrestaShopBundle/Command/UpdateSchemaCommand.php(76): PrestaShopBundle\Command\UpdateSchemaCommand->dropExistingForeignKeys(Object(Doctrine\DBAL\Connection), Object(Symfony\Component\Console\Output\BufferedOutput)) #1 vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php(255): PrestaShopBundle\Command\UpdateSchemaCommand->execute(Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\BufferedOutput)) #2 vendor/symfony/symfony/src/Symfony/Component/Console/Application.php(1039): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\BufferedOutput)) #3 vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php(97): Symfony\Component\Console\Application->doRunCommand(Object(PrestaShopBundle\Command\UpdateSchemaCommand), Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\BufferedOutput)) #4 vendor/symfony/symfony/src/Symfony/Component/Console/Application.php(275): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand(Object(PrestaShopBundle\Command\UpdateSchemaCommand), Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\BufferedOutput)) #5 vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php(83): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\BufferedOutput)) #6 vendor/symfony/symfony/src/Symfony/Component/Console/Application.php(149): Symfony\Bundle\FrameworkBundle\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\BufferedOutput)) #7 src/PrestaShopBundle/Service/Command/AbstractCommand.php(81): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\BufferedOutput)) #8 modules/autoupgrade/classes/UpgradeTools/SymfonyAdapter.php(56): PrestaShopBundle\Service\Command\AbstractCommand->execute() #9 modules/autoupgrade/classes/UpgradeTools/CoreUpgrader/CoreUpgrader80.php(68): PrestaShop\Module\AutoUpgrade\UpgradeTools\SymfonyAdapter->runSchemaUpgradeCommand() #10 modules/autoupgrade/classes/UpgradeTools/CoreUpgrader/CoreUpgrader.php(110): PrestaShop\Module\AutoUpgrade\UpgradeTools\CoreUpgrader\CoreUpgrader80->upgradeDb('8.1.0.0') #11 modules/autoupgrade/classes/UpgradeTools/CoreUpgrader/CoreUpgrader81.php(46): PrestaShop\Module\AutoUpgrade\UpgradeTools\CoreUpgrader\CoreUpgrader->doUpgrade() #12 modules/autoupgrade/classes/TaskRunner/Upgrade/UpgradeDb.php(43): PrestaShop\Module\AutoUpgrade\UpgradeTools\CoreUpgrader\CoreUpgrader81->doUpgrade() #13 admin995qvedjjpvqa7wcjtd/autoupgrade/ajax-upgradetab.php(53): PrestaShop\Module\AutoUpgrade\TaskRunner\Upgrade\UpgradeDb->run() #14 {main}`

### Expected behavior

I expected a successful update.

### Steps to reproduce

Go to autoupgrade module and update prestashop 8.1.0 to 8.1.1
Finally, when updating the database, I get an error.

### PrestaShop version(s) where the bug happened

8.1.0

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

ps_autoupgrade

**Comment by eveneme** [comment]
Changing to php 7.4 solved the problem, but I think it should be fixed, because it is not a long-term solution.

**Comment by letwang** [comment]
![image](https://github.com/PrestaShop/PrestaShop/assets/3644168/05577a75-f68f-41bf-b8d9-5b9a17649075)

**Comment by matks** [comment]
@eveneme In order to avoid this ticket to be closed like https://github.com/PrestaShop/PrestaShop/issues/32099 please provide a little more accurate informations 😉 

I give you an example:

1. Install PrestaShop [version X] on server with [configuration, including PHP version]
2. Install Upgrade module [version Y]
3. Go in backoffice page of Upgrade module, attempt to upgrade to [version Z]
4. Steps [...] go well
5. I see error [...]

**Comment by AureRita** [comment]
Hi @eveneme 

Thank you for your report, I tried to reproduce your issue with the provided information but I can't as you can see : 

[Untitled_ Aug 8, 2023 10_43 AM.webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/220c2f47-ffaf-4a54-a5b8-2d79bdc91411)

Did I miss something ? 

Can you give us the version of the module ? 
Can you provide a screen or a video to see the issue ? 

Waiting for your feedback

**Comment by kpodemski** [comment]
@eveneme 

My guess is that it is related to some of the modules you use which use Doctrine entities. Do you recall any?

**Comment by Hlavtox** [comment]
@kpodemski @matks We are adding something to an integer, which we are not really sure will be an integer I guess? https://github.com/PrestaShop/PrestaShop/blob/cd2f0c4495dfb0516a519932580a00259e7278ee/src/PrestaShopBundle/Command/UpdateSchemaCommand.php#L141

> In order to avoid this ticket to be closed like https://github.com/PrestaShop/PrestaShop/issues/32099 please provide a little more accurate informations 😉

I think there is no other information that could be provided - `Go to autoupgrade module and update prestashop 8.1.0 to 8.1.1`.

@eveneme Do you think you could help us here by running the upgrade again and checking what is happening?
Changing
`$nbQueries += $connection->executeQuery($drop);`
to something like
```
            file_put_contents( _PS_ROOT_DIR_ . 'testlog.txt', "NEWQUERY\n", FILE_APPEND);
            file_put_contents( _PS_ROOT_DIR_ . 'testlog.txt', "QUERY: " . $drop . "\n", FILE_APPEND);
            $r = $connection->executeQuery($drop);
            file_put_contents( _PS_ROOT_DIR_ . 'testlog.txt', "RESULT: " . print_r($r, true) . "\n\n\n", FILE_APPEND);
            $nbQueries += $r;
```
and checking what is happening?

**Comment by matks** [comment]
> I think there is no other information that could be provided - `Go to autoupgrade module and update prestashop 8.1.0 to 8.1.1`.

I did an upgrade from 8.1.0 to 8.1.1 5 days ago without issues, so I'm quite sur that there is much information to be provided 😋 . In other words, this quantity of infos is NOT enough for us to understand the exact scenario that can trigger the error.

**Comment by Hlavtox** [comment]
It's not enough to resolve why it fails, but definitely enough in order NOT to close the issue. I would open the same issue.

Thinking of what info could help us... @eveneme Could you also provide a dump of your SQL table structure? (No data)
![Snímek obrazovky 2023-08-08 134501](https://github.com/PrestaShop/PrestaShop/assets/6097524/64060b7b-eb86-452e-bcb1-d738768a69ac)

**Comment by ChillCode** [comment]
It's the same issue:

https://github.com/PrestaShop/PrestaShop/issues/32456

And easy to solve.

`$nbQueries += $connection->executeQuery($drop)->rowCount();`

**Comment by Hlavtox** [comment]
Yeah, but..
- Does this method even run normally? (Are there keys to drop?)
- If it runs, does this work? `$nbQueries += $connection->executeQuery($drop);`
- If it works, why there is something else returned for some executeQuery calls?

**Comment by ChillCode** [comment]
- This method (on my side) only fires when I use cli upgrade method, no when I use 1-click upgrade.
- On the other issue I explained why now PHP 8.1 throws an error.
- The result is the only thing I'm not sure what is expecting, before, casting an object would return always 1, now using ->rowcount() will return the total count.

**Comment by kpodemski** [comment]
@Hlavtox 

I think whether queries are executed or not depends on the database structure, but that's not the most important thing, I think @ChillCode is right, and explicitly returning a number of queries could do the job.

**Comment by Hlavtox** [comment]
@kpodemski @ChillCode Well, since we are counting `$nbQueries`, we could just increase it by 1.

```
$connection->executeQuery($drop);
$nbQueries++;
```

Update - but since the PHPDoc says `@return int The number of affected rows`, maybe it should really be `rowcount()` :-)

**Comment by kpodemski** [comment]
@Hlavtox to me it looks like a mistake in the code, it should've been rowCount since the beginning

**Comment by Hlavtox** [comment]
Will be fixed by https://github.com/PrestaShop/PrestaShop/pull/33573

**Comment by eveneme** [comment]
Server: Apache
PHP Version 8.1 and 8.0, 
Worked fine on 7.4.
The same error appeared when updating PrestaShop 8.0.x and 8.1. These are two different servers at a different service provider. I am getting this error during autoupgrade when updating the database. I always use the latest version of autoupgrade.

**Comment by eveneme** [comment]
> @eveneme
> 
> My guess is that it is related to some of the modules you use which use Doctrine entities. Do you recall any?

Standard installation without additional modules, just a template with its modules.

**Comment by hibatallahAouadni** [comment]
Fixed by #33573

**Comment by eveneme** [comment]
[INTERNAL] /var/www/vhosts/blanolove.pl/httpdocs/src/PrestaShopBundle/Command/UpdateSchemaCommand.php line 141 - TypeError: Unsupported operand types: int + Doctrine\DBAL\ForwardCompatibility\Result #0 /var/www/vhosts/blanolove.pl/httpdocs/src/PrestaShopBundle/Command/UpdateSchemaCommand.php(76): PrestaShopBundle\Command\UpdateSchemaCommand->dropExistingForeignKeys() #1 /var/www/vhosts/blanolove.pl/httpdocs/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php(255): PrestaShopBundle\Command\UpdateSchemaCommand->execute() #2 /var/www/vhosts/blanolove.pl/httpdocs/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php(1039): Symfony\Component\Console\Command\Command->run() #3 /var/www/vhosts/blanolove.pl/httpdocs/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php(97): Symfony\Component\Console\Application->doRunCommand() #4 /var/www/vhosts/blanolove.pl/httpdocs/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php(275): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() #5 /var/www/vhosts/blanolove.pl/httpdocs/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php(83): Symfony\Component\Console\Application->doRun() #6 /var/www/vhosts/blanolove.pl/httpdocs/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php(149): Symfony\Bundle\FrameworkBundle\Console\Application->doRun() #7 /var/www/vhosts/blanolove.pl/httpdocs/src/PrestaShopBundle/Service/Command/AbstractCommand.php(81): Symfony\Component\Console\Application->run() #8 /var/www/vhosts/blanolove.pl/httpdocs/modules/autoupgrade/classes/UpgradeTools/SymfonyAdapter.php(56): PrestaShopBundle\Service\Command\AbstractCommand->execute() #9 /var/www/vhosts/blanolove.pl/httpdocs/modules/autoupgrade/classes/UpgradeTools/CoreUpgrader/CoreUpgrader80.php(68): PrestaShop\Module\AutoUpgrade\UpgradeTools\SymfonyAdapter->runSchemaUpgradeCommand() #10 /var/www/vhosts/blanolove.pl/httpdocs/modules/autoupgrade/classes/UpgradeTools/CoreUpgrader/CoreUpgrader.php(110): PrestaShop\Module\AutoUpgrade\UpgradeTools\CoreUpgrader\CoreUpgrader80->upgradeDb() #11 /var/www/vhosts/blanolove.pl/httpdocs/modules/autoupgrade/classes/TaskRunner/Upgrade/UpgradeDb.php(43): PrestaShop\Module\AutoUpgrade\UpgradeTools\CoreUpgrader\CoreUpgrader->doUpgrade() #12 /var/www/vhosts/blanolove.pl/httpdocs/admin1965i5ry5datq5wbm2f/autoupgrade/ajax-upgradetab.php(53): PrestaShop\Module\AutoUpgrade\TaskRunner\Upgrade\UpgradeDb->run() #13 {main}
